HiringApp/HiringApp/Application/Scenes/ContactForm/ContactFormPresenter.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  ContactFormPresenter.swift
+//  HiringApp
+//
+//  Created by Alba Luján on 28/6/17.
+//  Copyright (c) 2017 Visual Engineering. All rights reserved.
+//
+
+import Foundation
+import BSWFoundation
+
+class ContactFormPresenter {
+
+    //MARK: - Stored properties
+    fileprivate let router: ContactFormRouterProtocol
+    fileprivate let interactor: ContactFormInteractorProtocol
+    fileprivate unowned let view: ContactFormUserInterfaceProtocol
+
+    var state: LoadingState<ContactFormViewModel> = .loading
+
+    var viewModel: ContactFormViewModel? {
+        didSet {
+            guard let viewModel = viewModel else {
+                state = .error(AppError.unknown)
+                return
+            }
+
+            state = .loaded(viewModel: viewModel)
+        }
+    }
+
+    //MARK: - Initializer
+    init(router: ContactFormRouterProtocol, interactor: ContactFormInteractorProtocol, view: ContactFormUserInterfaceProtocol) {
+        self.router = router
+        self.interactor = interactor
+        self.view = view
+    }
+    
+    //MARK: - Private API
+    fileprivate func sendContactFormData(candidate: ContactFormViewModel) {
+        interactor.sendContactFormData(candidate: candidate).upon(.main) { result in
+            switch result {
+            case .failure(let error):
+                self.state = .error(error)
+                showAlert(baseView: self.view as! UIViewController, title: "Error", message: "Error occured, please try again.")
+            case .success():
+                self.view.hideActivityIndicator()
+                self.router.navigateToNextScene()
+            }
+        }
+    }
+}
+
+extension ContactFormPresenter: ContactFormPresenterProtocol {
+
+    func viewDidLoad() {
+        self.viewModel = ContactFormViewModel()
+    }
+    
+    func tappedSendButton() {
+        guard let unwrappedViewModel = viewModel, self.viewModel?.validate() == true else {
```

Comment:
You can use the same variable name in cases like this one: 

`guard let viewModel = viewModel`